src/main/java/org/springframework/data/redis/cache/DefaultRedisCacheWriter.java

```diff
@@ -152,7 +163,11 @@ public byte[] putIfAbsent(String name, byte[] key, byte[] value, @Nullable Durat
 				if (shouldExpireWithin(ttl)) {
 					put = connection.set(key, value, Expiration.from(ttl), SetOption.ifAbsent());
 				} else {
-					put = connection.setNX(key, value);
+					if (shouldExpireMaxIdleWithin(maxIdle)) {
+						put = connection.set(key, value, Expiration.from(maxIdle), SetOption.ifAbsent());
+					} else {
+						put = connection.setNX(key, value);
+					}
```

Comment:
Again, handled in Spring Data Redis `3.0.x`, `3.1.x` (most recent GA version) and `3.2.x` (current development version).